src/facade/dragonfly_connection.cc

```diff
@@ -696,7 +696,12 @@ void Connection::OnBreakCb(int32_t mask) {
     return;  // we cancelled the poller, which means we do not need to break from anything.
 
   VLOG(1) << "Got event " << mask;
-  CHECK(cc_);
+
+  if (!cc_) {
+    LOG(ERROR) << "Unexpected event " << mask << " " << break_poll_id_;
```

Comment:
I have no context here, but don't we need to `notify()` listeners (like below) or whatnot? Or set our state to closed?

Reply:
you can look at the code where `breaker_cb_` is created and see yourself. Based on my readings there I think this should be fine.